### PR TITLE
update Miri

### PR DESCRIPTION
**Pull request by RalfJung**
r? @ghost

### CODE REVIEW

**Comment by oli-obk** [comment]
@bors r-

tests are failing due to

> error: failed to find or create the directory specified by `--out-dir`

that's a ui_test failure, huh. I'll check what's up

**Comment by oli-obk** [comment]
download-ci-llvm is broken... this is gonna take a bit

File: src/tools/miri/tests/compiletest.rs [hunk 1]
```diff
@@ -88,6 +88,7 @@ fn test_config(target: &str, path: &str, mode: Mode, with_dependencies: bool) ->
         mode,
         program,
         output_conflict_handling,
+        out_dir: std::env::var_os("CARGO_TARGET_DIR").unwrap().into(),
```
Comment (RalfJung) [inline src/tools/miri/tests/compiletest.rs:91]: Will this put random files into the out_dir? Maybe it should be a `miri` subdir of that folder, or so.

Why does this work in Miri CI, is there some kind of default?

Comment (oli-obk) [reply src/tools/miri/tests/compiletest.rs:91]: yea there's a default. I need to fix this in ui_test specifically (should always append `ui` or sth)

**Pull request merged**